pkg/build/reaper/reaper.go

```diff
@@ -43,6 +44,17 @@ func (reaper *BuildConfigReaper) Stop(namespace, name string, timeout time.Durat
 			return err
 		}
 
+		// Collect builds related to the config.
+		builds, err = reaper.oc.Builds(namespace).List(kapi.ListOptions{LabelSelector: buildutil.BuildConfigSelector(name)})
+		if err != nil {
+			return err
+		}
```

Comment:
and, due to the truncation issue, you need to filter the results using the annotation value, like the rest of the code does, to determine if a build is really related to the buildconfig.